pkg/clustermanager/cluster_manager.go

```diff
@@ -720,6 +720,12 @@ func (c *ClusterManager) EKSAClusterSpecChanged(ctx context.Context, clus *types
 
 func compareEKSAClusterSpec(ctx context.Context, currentClusterSpec, newClusterSpec *cluster.Spec) (bool, error) {
 	versions := newClusterSpec.Cluster.KubernetesVersions()
+	oldVersions := currentClusterSpec.Cluster.KubernetesVersions()
+
+	if len(versions) != len(oldVersions) {
```

Comment:
this makes it a bit more robust but still doesn't catch all cases
What if you had CP = 1.22 and worker1=1.21 and worker2=1.21 and then you upgrade to CP=1.22, worker1=1.22 and worker2=1.21

I advise to first write the tests so you can see where the wholes are

Reply:
The thing is that that EKSAClusterSpecChanged catches that case by comparing the old and new clusters. 
compareEKSAClusterSpec is only called by that function.